packages/lit-html/src/directives/template-content.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Directive, directive, noChange, NODE_PART, PartInfo} from '../lit-html.js';
+
+class TemplateContent extends Directive {
+  private __previousTemplate? : HTMLTemplateElement;
+  
+  constructor(part: PartInfo) {
+    super();
+    if (part.type !== NODE_PART) {
+      throw new Error('templateContent can only be used in text bindings');
+    }
+  }
+
+  render(template: HTMLTemplateElement) {
+    if (this.__previousTemplate === template) {
+      return noChange;
+    }
```

Comment:
Should we be supporting `noChange` and `nothing` values, too?

Reply:
Let me check in with the users who requested this. I suspect the more valuable value would be `null` since you might try to get a template out of the DOM and if it's not there the ref will be null.

We can add more value types at any time though, and this matches 1.x, so I'm ok with just this for now.